### ISSUE TITLE
XXX is assigned a value but never used  no-unused-vars

### ISSUE THREAD

**Issue by wang12124468** [eslint/eslint#13479]
**Tell us about your environment** 

* **ESLint Version:** 7.4
* **Node Version:** 10.15
* **npm Version:** *

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// eslint-disable-next-line no-unused-vars
let a = 1;
a = 2;
```

```bash

```

**What did you expect to happen?**

No lint for the second code(`a = 2`)

**What actually happened? Please include the actual, raw output from ESLint.**

Report a error for the code(`a = 2`) that `'a' is assigned a value but never used  no-unused-vars`

**Are you willing to submit a pull request to fix this bug?**

No

**Comment by anikethsaha** [comment]
Thanks for the issue

This is intended behavior.

here, the variable `a` still has only write a reference, no read reference. 
That is `a` is not being used, it is reassigned again.

**Comment by yeonjuan** [comment]
Hi, @wang12124468 

If you want to disable `no-unused-vars` using an `eslint-disable-next-line`, try it on the last reference. because it reports on the last.
* [online-demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgXCJuby11bnVzZWQtdmFyc1wiOiBcImVycm9yXCIgKi9cblxuXG5sZXQgYSA9IDE7XG5cbi8vIGVzbGludC1kaXNhYmxlLW5leHQtbGluZSBuby11bnVzZWQtdmFyc1xuYT0yIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJuby12YXIiOjJ9LCJlbnYiOnt9fX0=)

```js
/*eslint "no-unused-vars": "error" */


let a = 1;
a = 3;

// eslint-disable-next-line no-unused-vars
a=2;
```

**Comment by btmills** [comment]
Agreed this is working as intended. Hopefully the previous comments make sense for why we flag variables that may be assigned multiple times but are never read and how to use a disable line comment.

**Comment by wang12124468** [comment]
However, it is normal in version 4.x ?

**Comment by wang12124468** [comment]
I upgrade eslint 4.x -> 7.10, and other problems are resolved but only for this

**Comment by mdjermanovic** [comment]
@wang12124468 this was changed in v7.2.0.

It wasn't clear why `no-unused-vars` reports errors in a code where the variable is indeed used, but used only to update itself (e.g. `a = a + 1;`), so we decided to change the line where the rule reports error.

It's now the last line where the variable is assigned, which is `a = 2;` in your example.

`eslint-disable` comments are not aware of the context and the meaning of the reported error. They're simply filtering errors reported on the particular line.

That's why disabling the `let a = 1;` line has no effect anymore, because (as of v7.2.0) the rule reports error on the `a = 2;` line.

**Comment by wang12124468** [comment]
Got it. Thanks
